### PR TITLE
Fixes #160 - Skip signature without access_key_id and secret_access_key

### LINKED ISSUE #160: `AWS.STS.assume_role_with_web_identity` should not rely on access_key_id and secret_access_key
Calling `assume_role_with_web_identity` exchanges the token issued by OIDC provider for a temporary role credentials. The purpose of using the token is not to store the access_key_id and secret_access_key. However, it is currently not possible to achieve with aws-elixir. The `AWS.Request.request_post` tries to sign the headers with access_key_id and secret_access_key, which are not available. The signature is not needed and should be skipped.

Expected results:

I would like the following to work:
```elixir
alias AWS.STS
alias AWS.Client

STS.assume_role_with_web_identity(
  %Client{region: "us-west-1"},
  %{
    "WebIdentityToken" => "xyz",
    "RoleArn" => "my-role-arn",
    "RoleSessionName" => "my-session"
  }
)
```

Actual results:
** (FunctionClauseError) no function clause matching in :aws_signature.sign_v4/10
(aws_signature 0.3.1) /deps/aws_signature/src/aws_signature.erl:66: :aws_signature.sign_v4/10
(aws 0.13.2) lib/aws/request.ex:42: AWS.Request.request_post/5

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Review by onno-vos-dev — APPROVED** [review]
Will wait a day or so before merging to see if @philss (more Elixir experience) has any opinions on this :+1:

File: test/aws/request_test.exs [hunk 1]
```diff
@@ -4,43 +4,54 @@ defmodule AWS.RequestTest do
...
+    test "do not sign without access_key_id and secret_access_key" do
+      client = %Client{http_client: {TestClient, []}}
 
-        {status, _opts} = Keyword.pop(options, :return_status_code, 200)
+      assert {:ok, _response, _http_response} =
+               Request.request_post(client, @metadata, "Action", %{}, [])
 
-        {:ok, %{status_code: status, headers: [], body: "{\"Response\":\"foo\"}"}}
-      end
+      assert_receive {:request, :post, _url, _body, _headers, _options}
```
Comment (philss) [inline test/aws/request_test.exs:37]: I think you could match that the `headers` do not have the signature stuff. WDYT?

**Review by philss — COMMENTED** [review]
Overall looks good. Just a minor point :)

File: test/aws/request_test.exs (hunk 1 above)
Comment (xprazak2) [reply test/aws/request_test.exs]: Thanks, I added an assert for the signature not being in headers.

**Review by onno-vos-dev — APPROVED** [review]

**Comment by onno-vos-dev** [comment]
Thank you @xprazak2!

**Comment by onno-vos-dev** [comment]
@xprazak2 Tagged as v0.13.3 :+1: Available on hex.pm as well as Github Release :+1:

**Pull request merged**